components/webext-storage/sql/create_schema.sql

```diff
@@ -5,19 +5,20 @@
 -- This is a very simple schema.
 CREATE TABLE IF NOT EXISTS moz_extension_data (
     ext_id TEXT PRIMARY KEY,
-    /* The JSON payload. not null because we prefer to delete the row (possibly
-       creating a tombstone) than null it */
-    data TEXT NOT NULL,
+    -- The JSON payload. NULL means a tombstone (but see the CHECK below)
+    data TEXT,
 
-    /* Same "sync status" strategy used by other components. */
-    syncStatus INTEGER NOT NULL DEFAULT 0,
-    syncChangeCounter INTEGER NOT NULL DEFAULT 1
-) WITHOUT ROWID;
+    -- Same "sync status" strategy used by places components.
+    sync_status INTEGER NOT NULL DEFAULT 1, -- 1 == SyncStatus::New
+    sync_change_counter INTEGER NOT NULL DEFAULT 1,
 
-CREATE TABLE IF NOT EXISTS moz_extension_data_tombstones (
-    guid TEXT PRIMARY KEY
+    -- We only keep tombstones for items with a sync_status of Normal (and only
+    -- then until we upload the tombstone to the server. Enforce that.
+    CHECK(sync_status = 2 OR data IS NOT NULL)
 ) WITHOUT ROWID;
 
+/* index on sync_status? */
```

Comment:
That has the potential to be a very churny index, basically every write to this table will also need to update the index. It could be better on balance to pay the full table scan on each sync (not entirely sure how to test the two options in practice tho)